chugbot/attendance/roshHome.php

```diff
@@ -0,0 +1,279 @@
+                                <div id="date_pick" class="pb-2 ps-2">
+                                    <input type="date" id="start-date" name="start-date" class="form-control">
+                                </div>
+                            </div>
+                            <div class="col">
+                                <label class="description" for="end-date"><span style="color:red;">*</span>End Date (inclusive)</label>
+                                <div id="date_pick" class="pb-2">
+                                    <input type="date" id="end-date" name="end-date" class="form-control">
+                                </div>
+                            </div>
+                        </div>
+                    </li>
+                    <li style="margin:auto;" class="ps-0">
+                        <label class="description" for="edah"><span style="color:red;">*</span>Edah/Edot</label>
+                        <div id="edah_select" class="pb-2">
+                            <select class="form-select bg-info choices-js" id="edah_list" name="edah">
+                                <?php echo genPickList($edahId2Name, array(), "edah"); ?>
+                            </select>
+                        </div>
+                    </li>
+                    <li style="margin:auto;">
+                        <div class="row justify-content-center mt-2">
+                            <div class="col-6" style="text-align:center;"><button class="btn btn-primary" type="submit" id="submit_btn">View Attendance Matrix</button></div>
+                        </div>
+                    </li>
+                </ul></form>
+            </div>
+        </div>
+    </div>
+    </div>
+
+</div>
+
+<script>
+// automatically set date to "today"
+const today = new Date();
+const yyyy = today.getFullYear();
+const mm = String(today.getMonth() + 1).padStart(2, '0'); // Months start at 0!
+const dd = String(today.getDate()).padStart(2, '0');
+
+const formattedToday = `${yyyy}-${mm}-${dd}`;
+document.getElementById('date').value = formattedToday;
+
+
+function validateAttendanceForm() {
+    var error = "<div class=\"alert alert-danger alert-dismissible fade show mb-0 ms-2 me-2\" role=\"alert\">";
+    error += "<button type=\"button\" class=\"btn-close\" data-bs-dismiss=\"alert\" aria-label=\"Close\"></button>";
+    error += "<h5><strong>Error:</strong> Not all required fields are complete</h5><ul class=\"mb-0\">";
+
+    let valid = true;
+
+    // ensure each required field is filled out - first check field exists, then that it has a value
+    fields = ["date", "edah[]", "group"];
+    fields.forEach((field) => {
+        let x = document.forms["attendance_chug_select_form"][field];
+        if(field === "group") { field = "perek"; } // small override to keep backend and UI consistent with each other
```

Comment:
I don't quite understand this one. Should we change the frontend so the user inputs "perek" instead of "group"?

Reply:
Basically just a discrepancy about how we discuss it on the backend ("chug groups" = chug aleph, chug bet, etc) vs how we usually discuss it in conversation at camp ("prakim" = the time when we would have chug aleph, chug bet, etc). On the user's side, I used what we usually say at camp, but kept it consistent on the backend with how the practice has been thus far (especially with naming fields, database queries, etc). I wanted to keep it as consistent as possible on both sides, so added the small line which makes the switch from backend side to frontend view. I definitely didn't do the greatest job explaining it in the small one-line comment, but does that make a bit more sense?

Reply:
Ah, thank you- that makes sense! My main concern here is that some camp locations use different terms, and I didn't want to hard-code them too much. I think this is OK as a one-off alias ("perek" for "group"), but we may want to just expand the comment a little. I sometimes get confused myself about the difference between group and block in the database, and I wish I'd documented it better now!